packages/app/src/__tests__/DimensionMapper.test.tsx

```diff
@@ -20,25 +21,31 @@ test('display mapping for X axis when visualizing 2D dataset as Line', async ()
   expect(xDimsButtons[0]).not.toBeChecked();
   expect(xDimsButtons[1]).toBeChecked();
 
-  const d0Slider = screen.getByRole('slider');
+  // Ensure that only one dimension slider for D0 is visible
+  const d0Slider = screen.getByRole('slider', { name: 'D0' });
+  expect(d0Slider).toBeVisible();
   expect(d0Slider).toHaveAttribute('aria-valueNow', '0');
+  expect(screen.queryByRole('slider', { name: 'D1' })).not.toBeInTheDocument();
 
-  // Ensure that the swap from [0, 'x'] to ['x', 0] works
+  // Change mapping from [0, 'x'] to ['x', 0]
   await user.click(xDimsButtons[0]);
-
   await waitFor(() => expect(xDimsButtons[0]).toBeChecked());
   expect(xDimsButtons[1]).not.toBeChecked();
 
-  const d1Slider = screen.getByRole('slider');
+  // Ensure that the dimension slider is now for D1
+  const d1Slider = screen.getByRole('slider', { name: 'D1' });
+  expect(d1Slider).toBeVisible();
   expect(d1Slider).toHaveAttribute('aria-valueNow', '0');
+  expect(screen.queryByRole('slider', { name: 'D0' })).not.toBeInTheDocument();
 });
 
-test('display mappings for X and Y axes when visualizing 2D dataset as Heatmap', async () => {
-  const { user, selectVisTab } = await renderApp('/nD_datasets/twoD');
-
-  await selectVisTab(Vis.Heatmap);
+test('control mapping for X and Y axes when visualizing 2D dataset as Heatmap', async () => {
+  const { user } = await renderApp({
+    initialPath: '/nD_datasets/twoD',
+    preferredVis: Vis.Heatmap,
```

Comment:
Not sure if we want to set the preferredVis here. 

We want to rely on the fact that the `Heatmap` vis is the default vis for 2D datasets, no ? And let `preferredVis` be the focus of a dedicated test ?

Reply:
We're testing the `DimensionMapper` here, not the detection/selection of the supported visualizations (which is tested in `CorePack.test.ts`), and not the preferred visualization feature (which is tested in `VisSelector.test.ts`).